Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -320,30 +332,43 @@ def fetch_by_event_type(client: Client, event_type: EVENT_TYPE, events: dict, ma
-    last_fetch_time = f'{event_type.type}_last_fetch_time'
+    last_fetch_time_field = f'{event_type.type}_last_fetch_time'
+    last_fetch_next_field = f'{event_type.type}_last_fetch_next_field'
 
     demisto.debug(f'debug-log: handling event-type: {event_type.type}')
-    demisto.debug(f'debug-log: last run of type: {event_type.type} is: {last_run.get(last_fetch_time)}')
-    event_type_fetch_start_time = calculate_fetch_start_time(last_run.get(last_fetch_time), fetch_start_time)
-
-    response = client.fetch_by_aql_query(
+    if last_fetch_time := last_run.get(last_fetch_time_field):
+        demisto.debug(f'debug-log: last run of type: {event_type.type} time is: {last_fetch_time}')
+    last_fetch_next = last_run.get(last_fetch_next_field, 0)
+    demisto.debug(f'debug-log: last run of type: {event_type.type} next is: {last_fetch_next}')
+    event_type_fetch_start_time = calculate_fetch_start_time(last_fetch_time, fetch_start_time)
+    response, next = client.fetch_by_aql_query(
         aql_query=event_type.aql_query,
         max_fetch=max_fetch,
         after=event_type_fetch_start_time,
-        order_by=event_type.order_by
+        order_by=event_type.order_by,
+        from_param=last_fetch_next
     )
 
     demisto.debug(f'debug-log: fetched {len(response)} {event_type.type} from API')
     if response:
-        new_events, next_run[last_fetch_ids] = dedup_events(
-            response, last_run.get(last_fetch_ids, []), event_type.unique_id_key, event_type.order_by)
-        next_run[last_fetch_time] = new_events[-1].get(event_type.order_by) if new_events else last_run.get(last_fetch_time)
-        demisto.debug(f'debug-log: updated next_run with: {next_run[last_fetch_time]}')
-        events.setdefault(event_type.dataset_name, []).extend(new_events)
-        demisto.debug(f'debug-log: overall {len(new_events)} {event_type.dataset_name} (after dedup)')
-        demisto.debug(f'debug-log: last {event_type.dataset_name} in list: {new_events[-1] if new_events else {}}')
-    else:
-        next_run.update(last_run)
+        if last_fetch_next:
+            events.setdefault(event_type.dataset_name, []).extend(response)
+            demisto.debug(f'debug-log: last {event_type.dataset_name} in list: {response[-1]}')
+        else:
+            new_events, next_run[last_fetch_ids] = dedup_events(
+                response, last_run.get(last_fetch_ids, []), event_type.unique_id_key, event_type.order_by)
+            events.setdefault(event_type.dataset_name, []).extend(new_events)
+            demisto.debug(f'debug-log: overall {len(new_events)} {event_type.dataset_name} (after dedup)')
+            demisto.debug(f'debug-log: last {event_type.dataset_name} in list: {new_events[-1] if new_events else {}}')
+
+    if not next:  # we wish to update the time only in case the next is 0 because the next is relative to the time.
+        event_type_fetch_start_time = datetime.now().strftime(DATE_FORMAT)
```

Comment:
lets store time from last incident so we dont miss, 
this might make the case in the event that i started and ended in the same second that we are going over the list again, so we should so long as we dont advance the time we should store the ids.